File: HW13/src/main/java/ru/otus/web/service/AuthorService.java
package ru.otus.web.service;

import org.springframework.security.access.annotation.Secured;
import ru.otus.domain.Author;
import ru.otus.domain.Book;
import ru.otus.web.request.UpdateAuthorRequest;

import java.util.List;
import java.util.Optional;

public interface AuthorService {

	List<Author> getAllAuthors();

	Optional<Author> getAuthor(String name);

	List<Book> getAuthorBooks(Author author);

	@Secured("ROLE_ADMIN")
	Author updateAuthor(Author author, UpdateAuthorRequest request);
}

File: final-project/ui/src/services/AuthorService.js
// @flow
import RestClient from './RestClient';

import type { Author, AuthorFormValues } from 'types';

const BASE_PATH: string = '/book/author';

class AuthorService {

	static fetchAllAuthors(): Promise<Array<Author>> {
		return RestClient.get(BASE_PATH);
	}

	static updateAuthor(name: string, values: AuthorFormValues): Promise<void> {
		return RestClient.put(`${BASE_PATH}/${name}`, values)
	}
}

export default AuthorService;
